command_plugins/github/plugin.py

```diff
@@ -322,6 +330,66 @@ def add_outside_collab_command(self, data, user_data, collab, org, repo, permiss
                      markdown=True, thread=data["ts"])
 
     @hubcommander_command(
+        name="!SetRepoTeamPermissions",
+        usage="!SetRepoTeamPermissions <Team> <OrgWithRepo> <Repo> <Permission>",
+        description="This will set team permissions on a repository .",
+        required=[
+            dict(name="team", properties=dict(type=str, help="The team's name.")),
+            dict(name="org", properties=dict(type=str, help="The organization that contains the repo."),
+                 validation_func=lookup_real_org, validation_func_kwargs={}),
+            dict(name="repo", properties=dict(type=str, help="The repository to add the team to."),
+                 validation_func=extract_repo_name, validation_func_kwargs={}),
+            dict(name="permission", properties=dict(type=str.lower, help="The permission to grant, must be one "
+                                                                         "of: `{values}`"),
+                 choices="permitted_permissions")
+        ],
+        optional=[]
+    )
+    @auth()
+    @repo_must_exist()
+    def set_team_permissions(self, data, user_data, team, org, repo, permission):
+        """
+        Adds a team to a repository with a specified permission.
+
+        Command is as follows: !SetRepoPermissions <Team> <OrgWithRepo> <Repo> <Permission>
+        :param permission:
+        :param repo:
+        :param org:
+        :param team:
+        :param user_data:
+        :param data:
+        :return:
+        """
+        # Output that we are doing work:
+        send_info(data["channel"], "@{}: Working, Please wait...".format(user_data["name"]), thread=data["ts"])
+
+        # Check if team exists, if it does return the id
+        team_id = self.find_team_id_by_name(org, team)
```

Comment:
Bonus points (not required) if you make this a decorator :)

Reply:
You will need to add a check if the team doesn't exist.  Similar to: https://github.com/Netflix/hubcommander/blob/develop/command_plugins/github/plugin.py#L360-L362

Reply:
Hey, I've tried to make this part a decorator but I got myself into a corner and not sure what's the right way out.
My problem is that this part of the code not only checks if the team exists but also converts it from name to id. I'm not sure if this is a part of what the decorator should do, but since it's not doing it right now the function is being called twice which seems ineffective. Any help on this would be appreciated, but if it's too much to ask I'll just remove the decorator and push the other changes without it. Thanks!

Reply:
Decorator is not required -- just need to make sure that if the team doesn't exist, it handles the error gracefully.

Reply:
Wow, thank you so much, I was really busting my head on this.
Should work fine now.